addons/mass_mailing/views/mailing_contact_views.xml

```diff
@@ -43,7 +43,8 @@
         <field name="arch" type="xml">
             <tree string="Mailing List Contacts" sample="1" multi_edit="1" js_class="mailing_contact_list">
                 <header>
-                    <button name="action_import" string="Import" type="object" display="always"/>
+                    <button name="action_import" string="Import" type="object" display="always"
+                        context="{'from_mailing_list': context.get('active_model') == 'mailing.list', 'previous_active_ids': context.get('active_ids')}"/>
```

Comment:
Not sure "previous_active_ids" is very clear.

Could go with:

`'from_mailing_list_ids': context.get('active_ids') if context.get('active_model') == 'mailing.list' else False`

That way you only need one custom key?